ADR 454 for new product pages (#349)
alphagov/paas-team-manual: source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md

# ADR454: new product pages language and framework

## Context

[PaaS product pages] have been reviewed and a number of accessibility issues
have been identified. To resolve those we, would need to make an upgrade and
and review if any additional changes are needed to align with the GOV.UK Design System.

> As those pages are built on Ruby on Rails and in [ADR024] we've made the decision

Reviewer: ```suggestion
As those pages are built in Ruby and in [ADR024] we've made the decision
```

to develop our user-facing applications on Node, it's a good opportunity to 
look at rebuilding the product pages.

> We've disscussed user needs and it emerged that:

Reviewer: ```suggestion
We've discussed user needs and it emerged that:
```

* anyone in the team (developer and non-developer) should be able to update pages 
with less effort

> * pages should be performant for end users

Reviewer: ```suggestion
* pages should be performant for end users
* pages should be rendered by the server
```

Might be redundant with the preceding statement "performant" but also explicit

* pages should be rendered by the server

> * keeping pages up to date with GOV.UK Design System releases should be quiker and easier

Reviewer 2: ```suggestion
* keeping pages up to date with GOV.UK Design System releases should be quicker and easier
```

developer experience and give us the option to have shared component libraries

With the above in mind we researched options. Our admin interface is built in React,
so we narrowed the scope to React-based static site generators.

We ended up comparing two: [NextJS] with static page export and [GatsbyJS] 
(which exports static pages by default).
For page content we agreed that writing pages in [Markdown] is a good option,
so we tested both with [MDX] which can also embed React components inside content pages.

> [NextJS] and [GatsbyJS] have diffent approaches to development and there are minor 

Reviewer: ```suggestion
[NextJS] and [GatsbyJS] have different approaches to development and there are minor 
```

[NextJS] and [GatsbyJS] have different approaches to development and there are minor 
performance differences between them.

> Our usecase for now is narrow enough, and with the primary need of anyone in the team

Reviewer 2: ```suggestion
Our use case for now is narrow enough, and with the primary need of anyone in the team
```

## Decision

We will use [NextJS] together with [MDX] to author PaaS product pages content in 
[Markdown] and deliver them to users as static pages.

## Status

Accepted.

## Consequences

This implementation will allow us to:

- iterate our content more quickly as non-developers will be able to make chances easily

> - keep our page updated with newer releases of GOV.UK Design System to continue to makem

Reviewer: ```suggestion
- keep our page updated with newer releases of GOV.UK Design System to continue to make them
```
